### ISSUE TITLE
"mark" rule autocorrection is eating preceding comment content

### ISSUE THREAD

**Issue by voidless** [realm/SwiftLint#1749]
Swiftlint 0.21.0
"mark" rule autocorrection work incorrectly for commented out code, it removes all preceding comment content including comment open mark.
It should handle this case properly or ignore commented out code instead

Example:
```
/*
    func test1() {
    }
//MARK: mark
    func test2() {
    }
*/
```
Output:
```
// MARK: mark
    func test2() {
    }
*/
```

**Comment by marcelofabri** [comment]
I could reproduce it with 0.21.0.

**Comment by voidless** [comment]
I can still reproduce this on latest master (0.41.0) with by adding this corrections example to `MarkRule.swift` and rerunning tests
```swift
            Example("""
            /*
                func test1() {
                }
            ↓//MARK: mark
                func test2() {
                }
            */
            """): Example("""
            /*
                func test1() {
                }
            ↓// MARK: mark
                func test2() {
                }
            */
            """),
```

**Comment by goranche** [comment]
> I can still reproduce this on latest master (0.41.0) with by adding this corrections example to `MarkRule.swift` and rerunning tests

I hope you don't mind that I've used your example in the PR 😄
